sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+  bool version_found = false;
+  while (!(temp_found && version_found) || (jp.token.kind != AZ_JSON_TOKEN_END_OBJECT))
+  {
+    if (az_json_token_is_text_equal(&jp.token, desired_temp_property_name))
+    {
+      AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+      AZ_RETURN_IF_FAILED(az_json_token_get_uint32(&jp.token, parsed_value));
+      temp_found = true;
+    }
+    else if (az_json_token_is_text_equal(&jp.token, AZ_SPAN_FROM_STR("$version")))
+    {
+      AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+
+      AZ_RETURN_IF_FAILED(az_json_token_get_uint32(&jp.token, (uint32_t*)version_number));
+      version_found = true;
+    }
+    else
+    {
+      // else ignore token.
+      AZ_RETURN_IF_FAILED(az_json_parser_skip_children(&jp));
+    }
+    AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+  }
+  
+  if(temp_found && version_found)
+  {
+    return AZ_OK;
+  }
+
+  return AZ_ERROR_ITEM_NOT_FOUND;
+}
+
+static bool update_device_temp(int32_t temp)
+{
+  current_device_temp = temp;
+
+  bool ret = false;
+  if (current_device_temp > device_max_temp)
+  {
+    device_max_temp = current_device_temp;
+    ret = true;
+  }
+  if (current_device_temp < device_min_temp)
+  {
+    device_min_temp = current_device_temp;
+  }
+
+  // Increment the avg count, add the new temp to the total, and calculate the new avg
+  device_temperature_avg_count++;
+  device_temperature_avg_total += current_device_temp;
+  device_avg_temp = device_temperature_avg_total / (int32_t)device_temperature_avg_count;
+
+  return ret;
+}
+
+// Switch on the type of twin message and handle accordingly | On desired prop, respond with max
+// temp reported prop.
+static void handle_twin_message(
+    MQTTClient_message* message,
+    az_iot_hub_client_twin_response* twin_response)
+{
+  az_result result;
+  uint32_t desired_temp;
+  int32_t  version_num;
+  az_span twin_payload_span
+      = az_span_init((uint8_t*)message->payload, (int32_t)message->payloadlen);
+  // Determine what type of incoming twin message this is. Print relevant data for the message.
+  switch (twin_response->response_type)
+  {
+    // A response from a twin GET publish message with the twin document as a payload.
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_GET:
+      printf("A twin GET response was received\n");
+      if (az_failed(result =
+              parse_twin_desired_temperature_property(twin_payload_span, true, &desired_temp, &version_num)))
+      {
+        // If the item can't be found, the desired temp might not be set so take no action
+        break;
+      }
+      else
+      {
+        send_reported_temperature_property(desired_temp, version_num);
+
+        if (update_device_temp((int32_t)desired_temp))
+        {
+          send_reported_temperature_property(device_max_temp, version_num);
+        }
+      }
+      if (message->payloadlen)
+      {
+        printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
```

Comment:
>printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload); [](start = 8, length = 73)

nit: We should print the payload/payloadLen at top of function before we do anything with it.  OTherwise looking at a console the order stuff is happening isn't a clear.

Reply:
True. Payload print is now above the `switch`